.github/workflows/deploy.yml

```diff
@@ -0,0 +1,36 @@
+name: Deploy site
+
+on:
+  push:
+    branches:
+      - 'master'
+
+jobs:
+  deploy:
+    name: Deploy site to AWS S3
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: ${{ secrets.S3_BUCKET_REGION }}
+
+    - name: Build
+      run: mkdir dist/ && cp -a assets/ index.html dist/ && test -d dist/
```

Comment:
The `&& test -d dist/` might be redundant since it's in the same run line, but probably fine.